src/lib.rs

```diff
@@ -77,8 +77,14 @@ mod histogram;
 mod hybrid;
 mod squared_error;
 mod ssim;
+
+#[cfg(not(feature="yuv_compare"))]
 mod utils;
 
+/// Provides some utilities to make yuv image management and conversion easier.
+#[cfg(feature="yuv_compare")] // Exposes rgb/yuv conversions and split to yuv publicly, others were left to be pub crate or private
+pub mod utils;
+
```

Comment:
This behavior is completely new and should be covered by tests just to pin the api, as we do with all public API.

Reply:
`trait Decompose` is now covered by tests, `rgb_to_yuv` and `yuv_to_rgb` are covered by tests, which are currently the only 3 fully public APIs, the rest are crate-level publicity or private to the `utils` module. Compilation will fail should publicity be broken for these.

This is the closest that can be gotten for a test, and works since the cucumber test framework scripts can only access `mod utils` if the `yuv_compare` feature is enabled. There is sadly not a way to guarantee utils is private when it's disabled that i can find.